strax/context.py

```diff
@@ -1248,8 +1248,16 @@ def concat_loader(*args, **kwargs):
                 for dep_d in target_plugin.depends_on:
                     check_cache(dep_d)
 
-            # In case we can load the data already we want make a new superrun.
-            if loader and not (is_superrun and self.context_config["write_superruns"]):
```

Comment:
The most important bug is here. The exception of superrun is not needed. Because if the subruns are not available, the function will iteratively recursively call `check_cache`. So we do not have to worry about resaving superruns.